### PR TITLE
Auth: add the missing fields for all SSO providers

### PR DESCRIPTION
**Pull request by dmihai**
**What is this feature?**

Add all config fields for the SSO providers in the UI.

**Why do we need this feature?**

For SSO Settings API.

**Who is this feature for?**

everybody

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/auth-config/fields.tsx [hunk 1]
```diff
@@ -94,6 +80,130 @@ export const sectionFields: Section = {
...
+      id: 'general',
+      fields: ['name', 'clientId', 'clientSecret', 'scopes', 'allowSignUp', 'autoLogin', 'signoutRedirectUrl'],
+    },
+    {
+      name: 'User mapping',
+      id: 'user',
+      fields: ['allowAssignGrafanaAdmin', 'roleAttributePath', 'roleAttributeStrict', 'skipOrgRoleSync'],
+    },
+    {
+      name: 'Extra security measures',
```
[the remark this answers is not in the document]
Comment (dmihai) [reply public/app/features/auth-config/fields.tsx:128]: I added the new field `validate_hd`. Can you please check the description I added to it? https://github.com/grafana/grafana/blob/dmihai/sso-config-pages/public/app/features/auth-config/fields.tsx#L595-L600

File: public/app/features/auth-config/fields.tsx [hunk 2]
```diff
@@ -131,6 +113,132 @@ export const sectionFields: Section = {
...
+  google: [
+    {
+      name: 'General settings',
+      id: 'general',
+      fields: ['name', 'clientId', 'clientSecret', 'scopes', 'allowSignUp', 'autoLogin', 'signoutRedirectUrl'],
+    },
+    {
+      name: 'User mapping',
+      id: 'user',
+      fields: ['allowAssignGrafanaAdmin', 'roleAttributePath', 'roleAttributeStrict', 'skipOrgRoleSync'],
```
Comment (mgyongyosi) [inline public/app/features/auth-config/fields.tsx:125]: Please fix the order of these. The order should be (based on generic_oauth):  'roleAttributePath', 'roleAttributeStrict','allowAssignGrafanaAdmin','skipOrgRoleSync'.

File: public/app/features/auth-config/fields.tsx [hunk 3]
```diff
@@ -131,6 +113,132 @@ export const sectionFields: Section = {
...
+  gitlab: [
+    {
+      name: 'General settings',
+      id: 'general',
+      fields: [
+        'name',
+        'clientId',
+        'clientSecret',
+        'scopes',
+        'authStyle',
```
Comment (mgyongyosi) [inline public/app/features/auth-config/fields.tsx:180]: Not needed.

File: public/app/features/auth-config/fields.tsx [hunk 4]
```diff
@@ -131,6 +113,132 @@ export const sectionFields: Section = {
...
+  okta: [
+    {
+      name: 'General settings',
+      id: 'general',
+      fields: [
+        'name',
+        'clientId',
+        'clientSecret',
+        'scopes',
+        'authStyle',
```
Comment (mgyongyosi) [inline public/app/features/auth-config/fields.tsx:215]: Not needed.

File: public/app/features/auth-config/fields.tsx (hunk 2 above)
Comment (dmihai) [reply public/app/features/auth-config/fields.tsx]: Fixed this for all providers.

**Review by mgyongyosi — APPROVED** [review]
LGTM! Thanks! 🚀

**Pull request merged**